config/pipeline.yaml

```diff
@@ -812,19 +815,44 @@ build_variants:
     gcc-10:
       build_environment: gcc-10
       architectures:
-        x86_64:
+        x86_64: &x86_64_defconfig
           base_defconfig: 'x86_64_defconfig'
           filters:
             - regex: { defconfig: 'x86_64_defconfig' }
-        arm64:
+        arm64: &arm64_defconfig
           base_defconfig: 'defconfig'
           filters:
             - regex: { defconfig: 'defconfig' }
-        arm:
+        arm: &arm_defconfig
           base_defconfig: 'multi_v7_defconfig'
           filters:
             - regex: { defconfig: 'multi_v7_defconfig' }
 
+preempt_rt_variants:
+  variants: &preempt_rt_variants
+    gcc-10:
+      build_environment: gcc-10
+      # fragments:
+      #   - 'preempt_rt'
```

Comment:
`fragments` section looks like a part of `kbuild` job (in Maestro terms) definition, see https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/pipeline.yaml#L396

In other words: is not defined in the `build_config` nor `build_variants` but in a `job` definition for given `scheduler` entry.

See also `-chromeos` job definitions:
https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/jobs-chromeos.yaml#L14

Reply:
@pawiecz Still looking at it. 

One parallel question, How is `x86-board` fragment different from specifying `x86-board` again in extra_configs?
```
        x86_64:
          <<: *x86_64_defconfig
          fragments:
            - 'x86-board'
          extra_configs:
            - 'x86_64_defconfig+preempt_rt'
            - 'x86_64_defconfig+preempt_rt+x86-board'
```

Reply:
@pawiecz so the old way of specifying variants with config fragments doesn't work in the new KernelCI. I'm unable to figure out the new kbuild jobs. `kbuild-gcc-10-x86` uses a fragment. I'm not sure how it helps build up `baseline-x86`.

Reply:
> @pawiecz Still looking at it.
> 
> One parallel question, How is `x86-board` fragment different from specifying `x86-board` again in extra_configs?
> 
> ```
>         x86_64:
>           <<: *x86_64_defconfig
>           fragments:
>             - 'x86-board'
>           extra_configs:
>             - 'x86_64_defconfig+preempt_rt'
>             - 'x86_64_defconfig+preempt_rt+x86-board'
> ```

I see you're porting [this section from legacy config](https://github.com/kernelci/kernelci-core/blob/bad8ef4f39e5af434e359b817c74a6b0b1908a26/config/core/build-configs.yaml#L1038).

I'm not sure if `extra_configs` field is still respected in the new system - `extra_config` was [an argument to the legacy `kci_build` script](https://github.com/kernelci/kernelci-core/blob/bad8ef4f39e5af434e359b817c74a6b0b1908a26/kernelci/legacy/config/build.py#L154). I'm not able to provide you with an answer regarding the legacy but let's stick to the `fragments` section in the new system.

Reply:
> @pawiecz so the old way of specifying variants with config fragments doesn't work in the new KernelCI. I'm unable to figure out the new kbuild jobs. `kbuild-gcc-10-x86` uses a fragment. I'm not sure how it helps build up `baseline-x86`.

Right - there are several types of jobs that the new pipeline can execute. `kbuild-gcc-10-x86` (which indeed uses a fragment) is of a `kbuild` kind; see: https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/pipeline.yaml#L392 + https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/pipeline.yaml#L56 + https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/pipeline.yaml#L32

while `baseline-x86` is a job of a `test` kind; see: https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/pipeline.yaml#L216 + https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/pipeline.yaml#L17

In order to trigger any kind of a job you need to add a scheduler entry for it. `kbuild` kind of job is triggered by the changes made in trees defined in the `build_config` section, while `test` kind of a job is triggered based on the events defined in their scheduler entry: https://github.com/kernelci/kernelci-pipeline/blob/5ba096fdee5c99f1a04b60ee8e0a52b7677091ec/config/pipeline.yaml#L663

I hope that helps a bit to wrap one's head around this topic. I get that it might be confusing at first.

@JenySadadia wrote a nice primer on test enablement: https://github.com/kernelci/kernelci-pipeline/pull/649

Please let me know which topic was the hardest to grasp so that we can work on improving project's docs.